volumina/pixelpipeline/imagesources.py

```diff
@@ -689,30 +689,32 @@ def request( self, qrect, along_through=None ):
 from volumina.utility.segmentationEdgesItem import SegmentationEdgesItem, generate_path_items_for_labels
 class SegmentationEdgesItemSource(ImageSource):
     def __init__( self, layer, arraySource2D ):
-        from volumina.layer import SegmentationEdgesLayer
+        from volumina.layer import SegmentationEdgesLayer, LabelableSegmentationEdgesLayer
         assert isinstance(layer, SegmentationEdgesLayer)
         
         super( SegmentationEdgesItemSource, self ).__init__(layer.name)
         self._arraySource2D = arraySource2D
         self._arraySource2D.isDirty.connect(self.setDirty)
         self._layer = layer
+        self._is_clickable = isinstance(self._layer, LabelableSegmentationEdgesLayer)
 
     def request( self, qrect, along_through=None ):
         assert isinstance(qrect, QRect)
         # Widen request with a 1-pixel halo, to make sure edges on the tile borders are shown.
         qrect = QRect( qrect.x(), qrect.y(), qrect.width()+1, qrect.height()+1 )
         s = rect2slicing(qrect)
         arrayreq = self._arraySource2D.request(s, along_through)
-        return SegmentationEdgesItemRequest(arrayreq, self._layer, qrect)
+        return SegmentationEdgesItemRequest(arrayreq, self._layer, qrect, self._is_clickable)
 
     def image_type(self):
         return SegmentationEdgesItem
 
 class SegmentationEdgesItemRequest(object):
-    def __init__(self, arrayreq, layer, rect):
+    def __init__(self, arrayreq, layer, rect, is_clickable):
         self.rect = rect
         self._arrayreq = arrayreq
         self._layer = layer
+        self._is_clickable = is_clickable
```

Comment:
Here we can instead define and drop everything above if we define an interface (Layer.has_clickable_items) explicitly.
```python
@property
def _is_clickable(self):
      return self._layer.has_clickable_items
```